subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -328,14 +337,8 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
         run 'syncIt'
 
         then:
-        file('dest').assertHasDescendants(
-            'dir1/file1.txt',
-            'dir2/subdir/file2.txt',
-            'dir2/file3.txt',
-            'emptyDir'
-        )
-        file('dest/emptyDir').exists()
-        !file('dest/extra1.txt').exists()
+        file('dest/dir1/extra1.txt').exists()
+        file('dest/extraDir/extra1.txt').exists()
```

Comment:
❌ We accidentally removed some assertions.

```suggestion
        file('dest').assertHasDescendants(
            'dir1/file1.txt',
            'dir2/subdir/file2.txt',
            'dir2/file3.txt',
            'emptyDir'
            'dir1/extra1.txt',
            'extraDir/extra1.txt'
        )
```

Alternatively, we could restore the existing test and create a separate one with a `preserve {}` block.

(note: I've added the suggestion by hand; it needs testing)

Reply:
Readded old test and applied suggestions.